packages/netlify-cms-widget-relation/src/RelationControl.js

```diff
@@ -11,17 +11,18 @@ import { FixedSizeList } from 'react-window';
 const Option = ({ index, style, data }) => <div style={style}>{data.options[index]}</div>;
 
 const MenuList = props => {
-  if (props.isLoading || props.options.length <= 0) {
+  if (props.isLoading || props.options.length <= 0 || !Array.isArray(props.children)) {
     return props.children;
   }
   const rows = props.children;
+  const itemSize = 30;
   return (
     <FixedSizeList
       style={{ width: '100%' }}
       width={300}
-      height={300}
+      height={Math.min(300, rows.length * itemSize + itemSize / 3)}
```

Comment:
A small UI improvement to trim the height when there is a small number of entries.